packages/graphiql-react/src/editor/context.tsx

```diff
@@ -288,6 +312,7 @@ export function EditorContextProvider(props: EditorContextProviderProps) {
   useSynchronizeValue(queryEditor, props.query);
   useSynchronizeValue(responseEditor, props.response);
   useSynchronizeValue(variableEditor, props.variables);
+  useSynchronizeValue(extensionEditor, props.extensions);
```

Comment:
awesome stuff! i would prefer this to be possible via improvements to the plugin ecosystem, however `extensions` are already spec and 100% be a built-in feature of the execution context, though it will be tricky because as you can see there are many touchpoints. eventually each of these tabs will be provided by plugins, so someone could replace with a yaml and/or json schema react hook form implementation or some such

Reply:
Here's an example to illustrate:

say you have 4 tabs, each with the exact same query/variables/headers combination, but each with unique extension values. when you reload the browser, you'll only have one tab, because the persistence only picked up one (I haven't tried this but I assume this will happen b/c of the unique operation matching logic)